request/src/main/java/com/vimeo/networking2/VimeoApiClient.kt

```diff
@@ -1551,6 +1536,163 @@ interface VimeoApiClient {
+     * @return A [VimeoRequest] object to cancel API requests.
+     */
+    fun removeUserFromTeam(
+        membership: TeamMembership,
+        queryParams: Map<String, @JvmSuppressWildcards String>,
+        callback: VimeoCallback<Unit>
+    ): VimeoRequest
+
+    /**
+     * Changes the user role based on the given [role].
+     *
+     * @param uri the URI from which content will be sent to.
+     * @param role The [TeamRoleType] that given user will be changed to.
+     * @param queryParams Optional map used to refine the response from the API.
+     * @param callback The callback which will be notified of the request completion.
+     *
+     * @return A [VimeoRequest] object to cancel API requests.
+     */
+    fun changeUserRole(
+        uri: String,
+        role: TeamRoleType,
+        queryParams: Map<String, @JvmSuppressWildcards String>,
+        callback: VimeoCallback<TeamMembership>
+    ): VimeoRequest
+
+    /**
+     * Changes the user role based on the given [role].
+     *
+     * @param membership The [TeamMembership] of the User that will have their [TeamRoleType] changed.
+     * @param role The [TeamRoleType] that given user will be changed to.
+     * @param queryParams Optional map used to refine the response from the API.
+     * @param callback The callback which will be notified of the request completion.
+     *
+     * @return A [VimeoRequest] object to cancel API requests.
+     */
+    fun changeUserRole(
+        membership: TeamMembership,
+        role: TeamRoleType,
+        queryParams: Map<String, @JvmSuppressWildcards String>,
+        callback: VimeoCallback<TeamMembership>
+    ): VimeoRequest
+
+    /**
+     * Grants permission for the given users to access the given folder.
+     *
+     * @param uri the URI from which content will be sent to.
+     * @param usersIds A list of [GrantFolderPermissionForUser] containing URIs for users that will have access
+     * to the given folder, if a user who currently has access to this folder is not present in this list they will have
+     * their folder permission revoked.
+     * @param queryParams Optional map used to refine the response from the API.
+     * @param callback The callback which will be notified of the request completion.
+     *
+     * @return A [VimeoRequest] object to cancel API requests.
+     */
+    fun grantUsersAccessToFolder(
+        uri: String,
+        usersIds: List<GrantFolderPermissionForUser>,
+        queryParams: Map<String, @JvmSuppressWildcards String>,
+        callback: VimeoCallback<Unit>
+    ): VimeoRequest
+
+    /**
+     * Grants permission for the given users to access the given folder.
+     *
+     * @param folder The [Folder] that [usersIds] will be granted access to.
+     * @param usersIds A list of [GrantFolderPermissionForUser] containing URIs for users that will have access
+     * to the given folder, if a user who currently has access to this folder is not present in this list they will have
```

Comment:
This is a good callout.  

My initial thoughts are since the Edit/Create Folder screen is going to load all the team members anyway I can just filter that for users with access to this folder _but_ like you said for custom limits this might not work.  We might have to load the entire list of users recursively like suggested, I am curious though what the limit per response is since we had concerns about having to make recursive calls for the Analytics APIs but I was able to over 300+ data points in responses so I ended up not needing that. I'll have to look how iOS is currently implementing but regardless this will be a use case I'll have to keep in mind when I start working on this screen.

Reply:
Ah good point I forgot stats are separate